sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/tasks/TaskManagementSteps.java

```diff
@@ -96,6 +99,42 @@ public TaskManagementSteps(
           webDriverHelpers.clickOnWebElementBySelector(getByEventUuid(eventUuid));
         });
 
+    When(
+        "^I filter Task context by ([^\"]*)$",
+        (String filterType) -> {
+          webDriverHelpers.selectFromCombobox(TASK_CONTEXT_COMBOBOX, filterType);
+          webDriverHelpers.clickOnWebElementBySelector(APPLY_FILTER);
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(TABLE_ROW);
+        });
+
+    When(
+        "^I am checking if filter works correctly for ([^\"]*)$",
+        (String filterType) -> {
+          webDriverHelpers.checkWebElementContainsText(FILTER_ACTIVE_TASK_CONTEXT, filterType);
+        });
+
+    When(
+        "^I filter Task status ([^\"]*)$",
+        (String statusType) -> {
+          webDriverHelpers.selectFromCombobox(TASK_STATUS_COMBOBOX, statusType);
+          webDriverHelpers.clickOnWebElementBySelector(APPLY_FILTER);
+          webDriverHelpers.waitUntilIdentifiedElementIsPresent(TABLE_ROW);
+        });
+
+    When(
+        "^I am checking if filter works for ([^\"]*) status$",
+        (String statusType) -> {
+          webDriverHelpers.scrollToElementUntilIsVisible(FILTER_ACTIVE_TASK_STATUS);
+          webDriverHelpers.checkWebElementContainsText(FILTER_ACTIVE_TASK_STATUS, statusType);
+        });
+
+    When(
+        "I reset filter",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(RESET_FILTER);
+          TimeUnit.SECONDS.sleep(5);
```

Comment:
I know that due to application's UI development we're forced to use this wait, and I'm fine with it, but add also the wait for spinner to disappear, as I noticed that is visible and may take more than 5 seconds to have the page available to interact again